liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/ServerXmlDocument.java

```diff
@@ -0,0 +1,159 @@
+/**
+ * (C) Copyright IBM Corporation 2017.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package net.wasdev.wlp.maven.plugins;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.InputStream;
+import java.net.URL;
+import java.net.URLConnection;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+public class ServerXmlDocument {
+    
+    private ServerXmlDocument() {    
+    }
+    
+    public static boolean isFoundTagName(String fileName, String tag) throws Exception {
+
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document doc = builder.parse(fileName);
+        
+        NodeList appList = doc.getElementsByTagName(tag);
+        
+        return appList != null && appList.getLength() > 0;
+    }
+    
+    public static boolean isFoundWebApplication(String filePath, boolean bTarget) throws Exception {
+        
+        if (isFoundTagName(filePath, "webApplication"))
+            return true;
+         
+        boolean bFound = false;
+        
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Document doc = builder.parse(filePath);
+        NodeList appList = doc.getElementsByTagName("application");
+
+        if (isFoundWebAppElement(appList))
+            return true;
+
+        // check include elements against target server. 
+        if (bTarget) {
+            NodeList incList = doc.getElementsByTagName("include");
+        
+            if (incList == null)
+                return false;
+         
+            for (int i = 0; i < incList.getLength(); i++) {
+                Node locNode = incList.item(i).getAttributes().getNamedItem("location");
+                Document incDoc = null;
+                if (locNode != null && locNode.getNodeValue() != null && locNode.getNodeValue().endsWith(".xml")) {
+                    if (locNode.getNodeValue().startsWith("http")) {
```

Comment:
The include location attribute could be a file path or a URI for a remote resource.  I thought we don't handle remote resource (please confirm with the team). If we are handling remote resource, http is not the only protocol.